Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -10558,7 +10558,7 @@ WebPageCreationParameters WebPageProxy::creationParameters(WebProcessProxy& proc
     parameters.textAutosizingWidth = textAutosizingWidth();
     parameters.mimeTypesWithCustomContentProviders = protectedPageClient()->mimeTypesWithCustomContentProviders();
     parameters.deviceOrientation = m_deviceOrientation;
-    parameters.hardwareKeyboardState = internals().hardwareKeyboardState;
+    parameters.hardwareKeyboardState = m_legacyMainFrameProcess->processPool().cachedHardwareKeyboardState();
```

Comment:
Seems there is no guarantee that the cached hardware keyboard state has been initialized yet, right? Since it's happening on a background thread. I guess that's fine since we'll send IPC to the WebProcess once we actually cache the value the first time to let it know the value "changed"? Hopefully that's what's going on but I thought I would point it out so you make sure.

Reply:
That is correct. This patch relies on sending the keyboard state over to the WebContent process, once it is available.